src/onnx/parse_softmaxcrossentropyloss.cpp

```diff
@@ -0,0 +1,316 @@
+    Log probability tensor. If the output of softmax is prob, its value is log(prob).
+
+Type Constraints
+===================
+    T : tensor(float16), tensor(float), tensor(double), tensor(bfloat16) (Currently not supported in
+MIGX) Constrain input and output types to float tensors.
+
+Tind : tensor(int32), tensor(int64)
+    Constrain target to integer types
+*/
+
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/instruction.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct parse_softmaxcrossentropyloss : op_parser<parse_softmaxcrossentropyloss>
+{
+    std::vector<op_desc> operators() const { return {{"SoftmaxCrossEntropyLoss"}}; }
+
+    std::vector<instruction_ref> parse(const op_desc& /*opd */,
+                                       const onnx_parser& parser,
+                                       const onnx_parser::node_info& info,
+                                       const std::vector<instruction_ref>& args) const
+    {
+        // Get and handle attributes
+        std::string reduction = "mean";
+        if(contains(info.attributes, "reduction"))
+        {
+            std::set<std::string> supported_modes = {"mean", "none", "sum"};
+
+            reduction = info.attributes.at("reduction").s();
+            if(not contains(supported_modes, reduction))
+            {
+                MIGRAPHX_THROW("Invalid reduction mode: " + reduction +
+                               "\n Valid options are [none, mean, sum]");
+            }
+        }
+
+        // ignore_index is optional attribute, assign this as a scalar literal input to the op
+        auto ignore_index =
+            info.add_literal(migraphx::literal(migraphx::shape(shape::int64_type, {1}, {0}), {-1}));
+        bool has_ignore_index = contains(info.attributes, "ignore_index");
+        if(has_ignore_index)
+        {
+            auto ignore_index_val =
+                parser.parse_value(info.attributes.at("ignore_index")).at<int64_t>();
+            ignore_index = info.add_literal(migraphx::literal(
+                migraphx::shape(migraphx::shape::int64_type, {1}, {0}), {ignore_index_val}));
+        }
+
+        // Get and validate Inputs
+        auto scores = args.at(0);
+        auto labels = args.at(1);
+        auto scores_shape = scores->get_shape();
+        auto label_shape = labels->get_shape();
+
+        if(scores_shape.ndim() < 2)
+        {
+            MIGRAPHX_THROW("softmaxcrossentropyloss: Scores must be two or more dimensions [batch, "
+                           "class_size, D1...Dk]");
+        }
+
+        std::set<migraphx::shape::type_t> supported_label_types = {migraphx::shape::int32_type,
+                                                                   migraphx::shape::int64_type};
+        if(not contains(supported_label_types, label_shape.type()))
```

Comment:
This can be removed. The labels are used for input indices to scatter and gather and we support any type for those indices.

Reply:
Okay. Are we not worried about the inputs being to spec regardless if we can handle them? Should I just flag this as a warning in someway?

Reply:
Removed.

Reply:
> Are we not worried about the inputs being to spec regardless if we can handle them?

I dont see a problem with it, and if the spec is updated to support more types we dont need to make any changes.

Reply:
all good well Ive removed it so then we dont need to worry about it then. I think this means I can also remove the test case I added for the valid/invalid input